spark/dl/src/main/scala/org/apache/spark/ml/DLEstimator.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.ml
+import com.intel.analytics.bigdl.dataset.{DataSet, MiniBatch, Sample}
+import com.intel.analytics.bigdl.example.imageclassification.MlUtils._
+import com.intel.analytics.bigdl.{Criterion, Module}
+import com.intel.analytics.bigdl.optim.Optimizer
+import com.intel.analytics.bigdl.tensor.{Storage, Tensor}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import org.apache.spark.ml.param.{Param, ParamMap, Params}
+import org.apache.spark.sql.{DataFrame, Row}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+/**
+ * A wrapper of Optimizer to support fit() in ML Pipelines as an Estimator
```

Comment:
Since DataFrame only support array , we need to put the feature data , feature size, label data and label size as sub array of one array (thus 2-D array) and use the upper array as one column of datafram